src/main/java/com/comphenix/protocol/updater/SpigotUpdater.java

```diff
@@ -80,7 +80,12 @@ public void run() {
 			} finally {
 				// Invoke the listeners on the main thread
 				for (Runnable listener : listeners) {
-					plugin.getServer().getScheduler().scheduleSyncDelayedTask(plugin, listener);
+					try {
```

Comment:
Throwing exceptions as a means of control flow is generally discouraged because exceptions are designed to handle exceptional cases rather than regular control flow. Exceptions can be expensive in terms of performance and make code harder to read and maintain.

I recommend using a utility class that caches whether the server uses Folia. This way, you can use an if statement for control flow

Reply:
ok